src/main/handlers/status_check.py

```diff
@@ -1,45 +1,54 @@
-import requests
+import socket
 import asyncio
 
 from datetime import datetime
 
 import discord
-
-
-def get_status(data):
-    try:
-        map_name = data['map']
-        playercount = data['playerscount'][:data['playerscount'].index('/')]
-        game_str = f'{map_name} | {playercount} player' + 's' if playercount != '1' else ''
-
-        print(datetime.now().isoformat(), game_str)
-        return discord.Status.online, discord.Game(name=game_str)
-
-    except KeyError as error:
-        print(error, data)
-        if data['playerscount'] == 'offline':
-            return discord.Status.dnd, None
-
-    except BaseException as error:
-        print(error, data)
-        return discord.Status.invisible, None
+import steam.game_servers
 
 
 class StatusCheck:
 
-    def __init__(self, client):
+    def __init__(self, client, config):
         self.client = client
+        self.config = config
+
+        self.failed_checks = 0
+        self.maintenance = False
 
     async def execute(self):
-        # TODO stop using while True
-        while True:
-            print('\nChecking status...')
-
-            try:
-                data = requests.get('https://api.trackyserver.com/widget/index.php?id=300852').json()
-                status, activity = get_status(data)
-                await self.client.change_presence(status=status, activity=activity)
-            except BaseException as error:
-                print(datetime.now().isoformat(), error)
-            finally:
-                await asyncio.sleep(60)
+        try:
+            status, activity = self.get_status()
+            await self.client.change_presence(status=status, activity=activity)
+            print(f'{datetime.now().isoformat()} {status} - {activity}')
+        finally:
+            await asyncio.sleep(30)
+            self.client.loop.create_task(self.execute())
```

Comment:
Isn't it already looped somewhere else? https://github.com/MarkoDedovic/discord-bot/pull/5/files#diff-36a2e40a3fac2ba52a447e428d7cd3f7cd559b3797e6f25c198869cb6c0eea92R35